src/github/pullRequestModel.ts

```diff
@@ -997,6 +997,20 @@ export class PullRequestModel extends IssueModel<PullRequest> implements IPullRe
 			}
 		});
 
+		reviewThreads.forEach(thread => {
+			if (!thread.prReviewDatabaseId || !reviewEventsById[thread.prReviewDatabaseId]) {
+				return;
+			}
+			const prReviewThreadEvent = reviewEventsById[thread.prReviewDatabaseId];
+			prReviewThreadEvent.reviewThread = {
+				threadId: thread.id,
+				canResolve: thread.viewerCanResolve,
+				canUnresolve: thread.viewerCanUnresolve,
+				isResolved: thread.isResolved
+			};
+
+		});
+
```

Comment:
Why is this needed? The `PullRequestComments` query, which is what `reviewThreads` seems to come from, looks like it should have these properties already:

https://github.com/microsoft/vscode-pull-request-github/blob/da74def171b2bd0ff0123e015c3a67fb7fbe4ba8/src/github/queries.gql#L378-L386

Reply:
We are getting the information from `PullRequestComments`, but need the TimelineEvent to know about this information. And as the TimelineEvents API does not know about the threadId and resolve permissions, we need to add this information to the event.